TalismanSociety/signet-web
# Refactor/handle pruned blocks

# Description

Bittensor RPC only stores the latest 256 blocks, then they are pruned. An archive node is required In order to access the historical data of a given block older than the latest block - 256.

- Added `rpcUrlsOverride` to `chaingen`
- Handled pruned blocks errors.

## Type of change

Please delete options that are not relevant.


### 🧪 **How to test:**
- Execute a multisig tx on Bittensor
- Wait 256 blocks

Expected result:
Tx should still be displayed in Pending or Historical tabs.

### 🗒️ **Notes:**

- `REACT_APP_ON_FINALITY_API_KEY=myHeartIsInHavanaOohNaNa` must be added to `.env`
- ⚠️  API still going to be visible in the networks tabs on the browser, so don't forget to add API access restrictions!! ⚠️ 

### 🖼️ **Screenshots:**

![Screenshot 2024-07-25 at 18 11 46](https://github.com/user-attachments/assets/ec672087-c60f-48a0-891b-0ed896e55435)

![Screenshot 2024-07-25 at 18 11 38](https://github.com/user-attachments/assets/2117df8c-5bd6-4255-b0ab-aafffab6527c)
